frontend/dockerfile/dockerfile_buildinfo_test.go

```diff
@@ -0,0 +1,176 @@
+package dockerfile
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/containerd/continuity/fs/fstest"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+	"github.com/moby/buildkit/frontend/dockerfile/builder"
+	binfotypes "github.com/moby/buildkit/util/buildinfo/types"
+	"github.com/moby/buildkit/util/testutil/integration"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var buildinfoTests = integration.TestFuncs(
+	testBuildSources,
+	testBuildAttrs,
+)
+
+func init() {
+	allTests = append(allTests, buildinfoTests...)
+}
+
+// moby/buildkit#2311
+func testBuildSources(t *testing.T, sb integration.Sandbox) {
+	f := getFrontend(t, sb)
+
+	gitDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(gitDir)
+
+	dockerfile := `
+ARG DOCKERFILE_VERSION="1.3.0"
+FROM docker/dockerfile-upstream:${DOCKERFILE_VERSION} AS dockerfile
```

Comment:
We should avoid loading in new images from registry in tests. Only use the images that are already cached with the integration package mirror.